### PR TITLE
Gif documentation

### PR DESCRIPTION
**Pull request by davidsandilands**
I've created some gifs based off of our demos. I am wondering if they need re recorded with larger text.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -158,8 +158,39 @@ The CLI offers you a convinient way to create and retrieve data from the PDS. Yo
 ```
 pds-cli
 ```
-
 [The PDS CLI documentation](golang/pds-cli/doc/pds-cli.md) section has detailed instructions of the available options
+## PDS cli demo
```
Comment (mcka1n) [inline README.md:162]:
```suggestion
## PDS CLI demo
```

File: README.md [hunk 2]
```diff
@@ -158,8 +158,39 @@ The CLI offers you a convinient way to create and retrieve data from the PDS. Yo
...
+
+Demonstrating the PDS cli you can view the following gifs.
```
Comment (mcka1n) [inline README.md:164]: @davidsandilands  What do you think about this?

```suggestion
Demonstrating the PDS CLI capabilities with the below GIFs
```

File: README.md [hunk 3]
```diff
@@ -158,8 +158,39 @@ The CLI offers you a convinient way to create and retrieve data from the PDS. Yo
...
+
+1. Adding a class to a instance
```
Comment (mcka1n) [inline README.md:166]:
```suggestion
1. Adding a class to a node
```

File: README.md [hunk 4]
```diff
@@ -158,8 +158,39 @@ The CLI offers you a convinient way to create and retrieve data from the PDS. Yo
...
+
+Steps
+* Check the content of site.pp and hiera.yaml to allow for class and configuration on the server
+* Check the content of the client catalog on the client
+* Add the motd class to the client node with the pds-cli on the server
+* Running the puppet agent on the client node and confirm the new class is in the catalog and applied.
```
Comment (mcka1n) [inline README.md:172]: I wonder if we should ident "Steps" below the "Adding a class to a node" item?

```suggestion
    - Steps
       * Check the content of `site.pp` and `hiera.yaml` to allow for class and configuration on the server
       * Check the content of the client catalog on the client
      * Add the `motd` class to the client node with the `pds-cli` on the server
      * Running the Puppet agent on the client node and confirming the new class is in the catalog and applied.
```

File: README.md [hunk 5]
```diff
@@ -158,8 +158,39 @@ The CLI offers you a convinient way to create and retrieve data from the PDS. Yo
...
+
+![Add a class with PDS](./docs/images/pdsaddfact.gif)
+
+2. Adding hiera data to a node
+
+Steps
+* Add a hiera value for the client node for motd content 
+* Run puppet on the client and confirm the new content is pplied
```
Comment (mcka1n) [inline README.md:180]: Just suggesting the indentation as well

```suggestion
    - Steps
      * Add a Hiera value for the client node for `motd` content 
      * Run Puppet on the client and confirm the new content is applied
```

File: README.md [hunk 6]
```diff
@@ -158,8 +158,39 @@ The CLI offers you a convinient way to create and retrieve data from the PDS. Yo
...
+![Add hiera data with PDS](./docs/images/pdsaddhiera.gif)
+
+3. Adding a fact and using it in hiera data
+
+Steps
+* Add a fact to the client node with the pds-cli
+* Run puppet on the client to pick up the new fact
+* API call to show external facts on the server
+* Update the hiera content with the pds-cli to change motd content and include the fact
+* Run puppet on the client to apply the change to motd
```
Comment (mcka1n) [inline README.md:191]:
```suggestion

   - Steps
     * Add a fact to the client node with the `pds-cli`
     * Run Puppet on the client to pick up the new fact
     * API call to show external facts on the server
     * Update the Hiera content with the `pds-cli` to change `motd` content and include the fact
     * Run Puppet on the client to apply the change to `motd`
```

File: README.md [hunk 7]
```diff
@@ -158,8 +158,39 @@ The CLI offers you a convinient way to create and retrieve data from the PDS. Yo
...
+
+1. Adding a class to a instance
+
+Steps
+* Check the content of site.pp and hiera.yaml to allow for class and configuration on the server
+* Check the content of the client catalog on the client
+* Add the motd class to the client node with the pds-cli on the server
+* Running the puppet agent on the client node and confirm the new class is in the catalog and applied.
+
+![Add a class with PDS](./docs/images/pdsaddfact.gif)
```
Comment (mcka1n) [inline README.md:174]: I love the idea @davidsandilands 👍  great initiative! 

While the terminal/font size is small, it looks good when you Zoom in.

image

I wish it could go a bit slower though, because it's hard to follow it (lot's of changes in a short period of time), but it's doable if you watch the GIF a couple of times :)

Comment (davidsandilands) [reply README.md:174]: Let me have another shot with text size and see if i can get another gif recorder I was limited to 30 seconds

**Comment by reidmv** [comment]
Git hygiene thing: can we attach these files via [LFS](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github)? Make sure we aren't committing the binary images directly.

@davidsandilands I recognize that these files aren't crazy-big themselves, hence making this a hygiene comment more than an acute problem. I think using LFS for images/binaries is a good habit to get into, when we want to commit them to the repo. Ensures that the clone/download time doesn't get unduly impacted sometime in the far future by images or binaries that have long since become irrelevant or been replaced.
